packages/api/src/external/hie/__tests__/doc-progress-tests.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  Progress,
+  DocumentQueryStatus,
+  documentQueryStatus,
+} from "@metriport/core/domain/document-query";
+import { faker } from "@faker-js/faker";
+import { makeOptionalNumber } from "../../../shared/__tests__/utils";
+
+export const createProgress = ({
```

Comment:
The way this function works and is being used, it looks like its two functions combined in one. Maybe we split it, one to create progress based on status and another one based on progress?

If you think its better to have a single function, I'd at least suggest we try to use the progress param's numeric components to build total, error, and success, if available.